misk-hibernate-testing/src/main/kotlin/misk/hibernate/TruncateTablesService.kt

```diff
@@ -35,8 +38,11 @@ internal class TruncateTablesService(
   override val producedKeys = setOf<Key<*>>()
 
   override fun startUp() {
-    truncateUserTables()
-    executeStatements(startUpStatements, "startup")
+    // This is generally the first thing that happens in a test, if this fails it may be because Vitess isn't ready yet so we retry it a few times.
+    retry(10, ExponentialBackoff(Duration.ofMillis(20), Duration.ofMillis(1000))) {
```

Comment:
This is the wrong place. The service that sets up the Vitess connection should do this before it self-reports as healthy.

Reply:
Done. I added a PingDatabaseService that the SessionFactoryService depends on.